lib/authorizer/ntlm.js

```diff
@@ -241,7 +240,7 @@ module.exports = {
 
             // Now create the type 3 message, and add it to the request
             auth.set(NTLM_HEADER, authenticateMessage);
```

Comment:
Why keep `ntlm header` in the auth object, but state separately?

Reply:
Header data can be used in scripts, and it's an auth token, so storing this in the auth object makes sense.

Reply:
It feels like we're not sure of what data goes in `auth` object what data goes in `state`. If we're saying that only user data is in `auth` object, then this data does make sense in `state`.